lib/tarantool_server.py

```diff
@@ -569,37 +593,31 @@ def start(self, silent=True, wait=True, wait_load=True, rais=True,
         self.crash_detector = TestRunGreenlet(self.crash_detect)
         self.crash_detector.info = "Crash detector: %s" % self.process
         self.crash_detector.start()
-        wait = wait
-        wait_load = wait_load
-        if wait:
-            try:
-                self.wait_until_started(wait_load)
-            except TarantoolStartError:
-                # Raise exception when caller ask for it (e.g. in case of
-                # non-default servers)
-                if rais:
-                    raise
-                # Python tests expect we raise an exception when non-default
-                # server fails
-                if self.crash_expected:
-                    raise
-                if not self.current_test or not self.current_test.is_crash_reported:
-                    if self.current_test:
-                        self.current_test.is_crash_reported = True
-                    color_stdout('\n[Instance "{}"] Tarantool server failed to start\n'.format(
-                        self.name), schema='error')
-                    self.print_log(15)
-                # if the server fails before any test started, we should inform
-                # a caller by the exception
-                if not self.current_test:
-                    raise
-                self.kill_current_test()
+
+        if self.wait_for_start or wait_load:
+            self.wait_load(wait_load)
 
         port = self.admin.port
         self.admin.disconnect()
         self.admin = CON_SWITCH[self.tests_type]('localhost', port)
         self.status = 'started'
 
+    def wait_load(self, wait_load):
+        try:
+            self.wait_until_started(wait_load)
+        except Exception:
```

Comment:
I think we should avoid 'except Exception' (at least w/o proper comment what going on). There are few errors can be occured here. The code should expect only that errors.